tasks/go.py

```diff
@@ -428,3 +435,182 @@ def go_fix(ctx, fix=None):
+        deps = [
+            dep.removeprefix(prefix)
+            for dep in deps
+            if dep.startswith(prefix) and dep != f'github.com/DataDog/datadog-agent/{path}'
+        ]
+
+        return deps
+
+
+def add_replaces(ctx, path, replaces: list[str]):
+    repo_path = os.path.abspath('.')
+    with ctx.cd(path):
+        for repo_local_path in replaces:
+            if repo_local_path != path:
+                # Example for pkg/util/log with path=pkg/util/cachedfetch
+                # - repo_local_path: pkg/util/log
+                # - online_path: github.com/DataDog/datadog-agent/pkg/util/log
+                # - module_local_path: ../../../pkg/util/log
+                level = os.path.abspath(path).count('/') - repo_path.count('/')
+                module_local_path = ('./' if level == 0 else '../' * level) + repo_local_path
+                online_path = f'github.com/DataDog/datadog-agent/{repo_local_path}'
+
+                ctx.run(f"go mod edit -replace={online_path}={module_local_path}")
+
+
+def add_go_module(path):
+    """
+    Add go module to modules.py
+    """
+    print(color_message("Updating DEFAULT_MODULES within modules.py", "blue"))
+    modules_path = tasks.modules.__file__
+    with open(modules_path) as f:
+        modulespy = f.read()
+
+    modulespy_regex = re.compile(r"DEFAULT_MODULES = {\n(.+?)\n}", re.DOTALL | re.MULTILINE)
+
+    all_modules_match = modulespy_regex.search(modulespy)
+    all_modules = all_modules_match.group(1)
+    all_modules = all_modules.split('\n')
+    indent = ' ' * 4
+
+    new_module = f'{indent}"{path}": GoModule("{path}", independent=True),'
+
+    # Insert in order
+    insert_line = 0
+    for i, line in enumerate(all_modules):
+        # This line is the start of a module (not a comment / middle of a module declaration)
+        if line.startswith(f'{indent}"'):
+            module = re.search(rf'{indent}"([^"]*)"', line).group(1)
+            if module < path:
+                insert_line = i
+            else:
+                assert module != path, f"Module {path} already exists within {modules_path}"
+
+    all_modules.insert(insert_line, new_module)
+    all_modules = '\n'.join(all_modules)
+    with open(modules_path, 'w') as f:
+        f.write(modulespy.replace(all_modules_match.group(1), all_modules))
+
+
+@task
+def create_module(ctx, path: str, no_verify: bool = False):
+    """
+    Create new go module following steps within <docs/dev/modules.md>
+    - packages: Comma separated list of packages the will use the new module
+    """
+
+    path = path.rstrip('/')
+
+    if check_uncommitted_changes(ctx):
+        raise RuntimeError("There are uncomitted changes, all changes must be committed to run this command.")
+
+    # Perform checks + save current state to restore it in case of failure
+    assert not os.path.exists(path + '/go.mod'), f"Path {path + '/go.mod'} already exists"
+    is_empty = not os.path.exists(path)
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏Have you checked Windows ?